R/config-log.R

```diff
@@ -0,0 +1,129 @@
+###############################
+# Fields from bbi_config.json
+###############################
+
+#' Parse babylon configs to log
+#'
+#' Parses `bbi_config.json` files into a log tibble.
+#' This file is created by babylon, in the model output folder.
+#' It stores metadata about the execution of a model run,
+#' including md5 hashes of the model file (control stream) and data file at the time the model was run.
+#'
+#' @details
+#' `config_log()` will return a tibble with one row per `bbi_config.json` found.
+#'
+#' `add_config()` takes a `bbi_run_log_df` tibble (the output of `run_log()`) as its input,
+#' searches for a `bbi_config.json` for each row in the input tibble,
+#' and joins on the data from the relevant config, if found.
+#' The returned tibble will have all of its input columns, plus all of the columns returned from `config_log()`
+#'
+#' @seealso `run_log()`
+#' @param .base_dir Base directory to look in for models. Defaults to `get_model_directory()`, and falls back to `getwd()` if `get_model_directory()` returns `NULL`.
+#' @param .recurse If `TRUE`, the default, search recursively in subdirectories.
+#' @export
+config_log <- function(
+  .base_dir = get_model_directory(),
+  .recurse = TRUE
+) {
+
+  # if no directory defined, set to working directory
+  if (is.null(.base_dir)) {
+    .base_dir <- getwd()
+  }
+
+  mod_list <- find_models(.base_dir, .recurse)
+  if(is.null(mod_list)) {
+    return(NULL)
```

Comment:
I can't think of a good reason to return `NULL` instead of an empty tibble. One question though: is it good enough to just return `tibble()` or do you think it should have the same columns as the successful tibble, but with zero rows? Obviously the former would be easier, but in the back of my mind something is telling me to consider the latter.

Reply:
In terms of an error... I don't know. That just feels a little bit aggressive. I can see the argument that why would anyone want to silently be returned an empty tibble, but my gut tells me this is the better pattern. Do you feel strongly?

Reply:
Discussed on the phone, but yes I will make the result an empty tibble. Also, this is a somewhat relevant to https://github.com/metrumresearchgroup/rbabylon/pull/118/files#r466540698